controller/src/beerocks/master/db/node.h

```diff
@@ -182,6 +186,26 @@ class node {
     };
     std::shared_ptr<radio> hostap;
 
+    class link_metrics_data{
+        public:
+        
+        link_metrics_data(){};
+        ~link_metrics_data(){};
+        
+        sMacAddr al_mac_of_the_device_that_is_reported;
+        std::list<ieee1905_1::tlvTransmitterLinkMetric::sInterfacePairInfo> transmitterLinkMetrics;
+        std::list<ieee1905_1::tlvReceiverLinkMetric::sInterfacePairInfo> receiverLinkMetrics;
+
+        bool add_transmitter_link_metric(std::shared_ptr<ieee1905_1::tlvTransmitterLinkMetric> TxLinkMetricData);
+        bool add_reciever_link_metric(std::shared_ptr<ieee1905_1::tlvReceiverLinkMetric> RxLinkMetricData);
+    };
+    /*
+    * This map holds link metric data per Agent.
+    * Created empty in all nodes
+    * Used only in GW node (retreave from DB method get_metric_data_map)
+    */
+    std::map<std::string, son::node::link_metrics_data> mOfMetricData;
```

Comment:
This is not correct: now you can have only one neighbour per agent. You need a list of link_metrics_data.

Since we get the transmit and receive metrics separately (in separate TLVs), it's probably easier to have a completely separate map of  `agent1 -> list<(agent2, vector<interface_pair_tx_link_metric>)>` and `agent1 -> list<(agent2, vector<interface_pair_rx_link_metric>)>`.

Also, I see no reason at all why `agent1` can't be specified with a sMacAddr instead of as a string.